src/modules/leetcode.ts

```diff
@@ -0,0 +1,240 @@
+    context: "leetcode.initializeForeground",
+  });
+
+  chrome.runtime.onMessage.addListener(async (message: SubmissionTrigger) => {
+    switch (message.type) {
+      case "submission-trigger":
+        const { submissionId } = message.payload;
+        const details = await getSubmissionDetails(submissionId);
+        const { statusCode } = details;
+        if (statusCode !== 10) {
+          break;
+        }
+        const { titleSlug } = details.question;
+        const topicTags = await getQuestionTopics(titleSlug);
+        const newMessage: SubmissionHandler = {
+          type: "submission-handler",
+          payload: { details, topicTags },
+        };
+        chrome.runtime.sendMessage(newMessage);
+        break;
+    }
+    return true;
+  });
+};
+
+/**
+ * Initialize the leetcode module, for the background script
+ * @param octokit The Octokit instance
+ */
+export const initializeBackground = async (octokit: Octokit): Promise<void> => {
+  await debugReport({
+    message: "Initialize the LeetCode module, for the background script",
+    context: "leetcode.initializeBackground",
+  });
+
+  const tabs = await chrome.tabs.query({ url: "https://leetcode.com/*" });
+  for (const tab of tabs) {
+    chrome.tabs.reload(tab.id);
+  }
+
+  chrome.webRequest.onBeforeRequest.addListener(
+    (details) => {
+      const match = details.url.match(/detail\/(.*?)\/check/);
+      submitted.add(parseInt(match[1]));
+    },
+    { urls: ["https://leetcode.com/submissions/detail/*/check/"] }
+  );
+
+  chrome.webRequest.onBeforeRequest.addListener(
+    (details) => {
+      const encoded = details.requestBody.raw[0].bytes;
+      const decoder = new TextDecoder("utf-8");
+      const decoded = decoder.decode(encoded);
+      const request: GraghQueryRequest = JSON.parse(decoded);
+      if (request.operationName == "submissionDetails") {
+        const { submissionId } = request.variables;
+        if (submitted.delete(submissionId)) {
+          const message: SubmissionTrigger = {
+            type: "submission-trigger",
+            payload: { submissionId },
+          };
+          chrome.tabs.sendMessage(details.tabId, message);
+        }
+      }
+    },
+    { urls: ["https://leetcode.com/graphql/"] },
+    ["requestBody"]
+  );
+
+  chrome.runtime.onMessage.addListener(async (message: SubmissionHandler) => {
+    switch (message.type) {
+      case "submission-handler":
+        const { details } = message.payload;
+        const prefix = details.question.titleSlug;
+        const suffix = extensionLookup[details.lang.name];
+        const payload: FileToCommit[] = [];
+        for (const tag of message.payload.topicTags) {
+          payload.push({
+            path: `${tag.slug}/${prefix}.${suffix}`,
+            content: message.payload.details.code,
+          });
+        }
+        const commitMessage = [
+          `:ballot_box_with_check: LC-${details.question.titleSlug}`,
```

Comment:
Can we change back to brackets? Its easier to differentiate sections.

Reply:
Fixed :)